addons/l10n_ch/models/account_invoice.py

```diff
@@ -294,18 +316,20 @@ def isr_print(self):
                                    - fill the 'bank account' field of the invoice with the postal to be used to receive the related payment. A default account will be automatically set for all invoices created after you defined a postal account for your company."""))
 
     def print_ch_qr_bill(self):
-        """ Triggered by the 'Print QR-bill' button.
+        """
+        Triggered by the 'Print QR-bill' button.
         """
         self.ensure_one()
 
         if not self.partner_bank_id:
             raise UserError(_("QR-Bill can not be generated on paid invoices. If the invoice is not fully paid, please make sure Recipient Bank field is not empty and try again."))
-
-        if not self.partner_bank_id._eligible_for_qr_code('ch_qr', self.partner_id, self.currency_id):
-            raise UserError(_("Cannot generate the QR-bill. Please check you have configured the address of your company and debtor. If you are using a QR-IBAN, also check the invoice's payment reference is a QR reference."))
+        if not self.partner_bank_id._eligible_for_qr_code('ch_qr', self.partner_id, self.currency_id, False):
```

Comment:
add the keyword `raise_errors` plz. Then if someone add an extra optional parameter, this will not be the zbeul everywhere :)

Reply:
You're right, the zbeul should be avoided at all cost :)